### PR TITLE
Hybrid ce loss changes

### PR DESCRIPTION
**Pull request by zhiyil-graphcore**
## Changelogs

- _enumerate the changes of that PR._

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation is a new function is added, or an existing one is deleted._
- [ ] _Write concise and explanatory changelogs above._
- [ ] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

Fix to hybridceloss, but it is not actually helping the hybridceloss for val and test using the debug config.

### CODE REVIEW

File: graphium/trainer/losses.py [hunk 1]
```diff
@@ -59,10 +60,14 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
         """
 
         target = target.flatten()
-
-        regression_input = torch.inner(input, self.brackets.to(input.device))
+        # regression loss needs normalized logits to probability as input to do inner product with self.brackets
+        # we apply softmax on the raw logits first
+        softmax_input = self.softmax(input)
```
Comment (DomInvivo) [inline graphium/trainer/losses.py:65]: I thought we discussed that the `softmax` should go as the `last_activation` since we want the classification metrics to work

**Review by DomInvivo — CHANGES_REQUESTED** [review]

File: graphium/trainer/losses.py (hunk 1 above)
Comment (zhiyil-graphcore) [reply graphium/trainer/losses.py:65]: I mentioned about this on [slack](https://graphcore.slack.com/archives/C03NG9SA3QC/p1687515389789519) about the auroc and average precision metrics. It looks like they are happy with both probabilities or logits. In that case we don't need a custom CE loss? Or do you think there would be other classification metrics which only accept probabilities as the input?
```
`    Accepts the following input tensors:

    - ``preds`` (float tensor): ``(N, C, ...)``. Preds should be a tensor containing probabilities or logits for each
      observation. If preds has values outside [0,1] range we consider the input to be logits and will auto apply
      softmax per sample.`
```

Comment (DomInvivo) [reply graphium/trainer/losses.py:65]: Thanks, just saw your Slack message!

I checked also Accuracy and f1. I think this is consistent across metrics 🎉
